kolibri_zim_plugin/assets/src/views/ZimSearchView.vue

```diff
@@ -67,38 +91,80 @@
     data() {
       return {
         isSearchWaiting: 0,
+        isSearchMoreWaiting: 0,
         searchResults: {},
       };
     },
-    computed: {},
+    computed: {
+      searchHasMore() {
+        return this.searchResults.next;
+      },
+      remainingResults() {
+        if (this.searchResults.next) {
+          return this.searchResults.count - this.searchResults.next;
+        } else {
+          return 0;
+        }
+      },
+    },
     methods: {
       /**
        * @public
        */
       focus() {
         this.$refs.searchForm.focus();
       },
-      onSearchFormSubmit(query) {
-        this.startSearch(query);
-      },
       onSearchFormReset() {
         this.searchResults = {};
       },
-      startSearch(query) {
+      onSearchFormSubmit(query) {
         this.isSearchWaiting += 1;
-
-        const max_results = 100;
-
-        ZimSearchResource.search(this.zimFilename, { query, max_results })
+        this.startSearch(query).finally(() => {
+          this.isSearchWaiting -= 1;
+        });
+      },
+      onSearchMoreClick() {
+        this.isSearchMoreWaiting += 1;
+        this.startSearch(this.searchResults.query, this.searchResults.next).finally(() => {
+          this.isSearchMoreWaiting -= 1;
+        });
+      },
+      startSearch(query, start = 0) {
+        const max_results = 10;
+        const snippet_length = 140;
+
+        return ZimSearchResource.search(this.zimFilename, {
+          query,
+          start,
+          max_results,
+          snippet_length,
+        })
           .then(result => {
             const { articles, count } = result.data;
-            this.searchResults = { query, articles, count, success: true };
+            const end = start + articles.length;
+            const next = end < count ? end : null;
+
+            if (query === this.searchResults.query && start == this.searchResults.next) {
```

Comment:
the second comparison uses `==` is that intentional or it's just a typo? Maybe it's a good idea to add some kind of linting with eslint.

Reply:
Good spotting! This does use eslint, so it's funny that it didn't complain about that :o